parachain/pallets/polkaerc20-app/src/lib.rs

```diff
@@ -3,60 +3,47 @@
 /// Implementation for PolkaERC20 token assets
 ///
 use sp_std::prelude::*;
-use sp_std::{fmt::Debug, convert::TryInto};
-use sp_core::H160;
-use sp_runtime::traits::{
-	CheckedAdd, MaybeSerializeDeserialize, Member,  AtLeast32BitUnsigned};
-use frame_system::{self as system};
+use sp_core::{H160, U256, RuntimeDebug};
+use frame_system::{self as system, ensure_signed};
 use frame_support::{
-	decl_error, decl_event, decl_module, decl_storage, Parameter,
+	decl_error, decl_event, decl_module, decl_storage,
 	dispatch::{DispatchResult, DispatchError},
-	storage::StorageDoubleMap
 };
 
-use codec::{Decode};
+use codec::{Encode, Decode};
 
-use artemis_core::{AppID, Application, Message};
+use artemis_core::{AppID, Application, RelayEventEmitter, Message};
 use artemis_ethereum::{self as ethereum, SignedMessage};
+use artemis_generic_asset as generic_asset;
 
 #[cfg(test)]
 mod mock;
 
 #[cfg(test)]
 mod tests;
 
-pub trait Trait: system::Trait {
-	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+pub const APP_ID: &[u8; 32] = &[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
```

Comment:
I’d there a reason why we use 1 here? We could also hash „PolkaERC20“ (or similar) to prevent conflicting ids, right?

Reply:
No specific reason. Though while we're on the topic, I'm not sure why we still need AppID's to be 256 bits. We originally thought we needed the whole 256 bits so we could stuff a contract address into it. But that hasn't been necessary to make our PolkaERC20 app work. In fact currently AppIDs are not used at all in the current parachain.

Reply:
Happy to shorten it, which length makes sense to you? 64 bits?

Reply:
We'll tackle AppId's in M5.

Reply:
@vgeddes please add github issue or trello task